File: backend/MiniOrder.Domain/Entities/Category.cs
namespace MiniOrder.Domain.Entities;

public class Category
{
    public int Id { get; set; }

    public string Name { get; set; } = string.Empty;

    public string Slug { get; set; } = string.Empty;

    public bool IsActive { get; set; } = true;

    public int? ParentCategoryId { get; set; }

    public Category? ParentCategory { get; set; }

    public ICollection<Category> SubCategories { get; set; } = new List<Category>();

    public ICollection<Product> Products { get; set; } = new List<Product>();

    public ICollection<CategoryAttribute> Attributes { get; set; } = new List<CategoryAttribute>();
}

File: frontend/src/types/category.ts
export interface SubCategory {
  id: number;
  name: string;
  slug: string;
}

export interface Category {
  id: number;
  name: string;
  slug: string;
  subCategories: SubCategory[];
}

export interface CategoryDetail {
  id: number;
  name: string;
  slug: string;
  isActive: boolean;
  parentCategoryId: number | null;
  parentCategoryName: string | null;
  subCategories: SubCategory[];
}

export interface CreateCategoryRequest {
  name: string;
  slug: string;
  parentCategoryId: number | null;
}

export interface UpdateCategoryRequest {
  name: string;
  slug: string;
  parentCategoryId: number | null;
  isActive: boolean;
}
